pkg/backend/httpstate/journal/snapshot.go

```diff
@@ -139,41 +139,66 @@ func sendBatch(
 	return sendBatch(ctx, client, update, tokenSource, batch[len(batch)/2:])
 }
 
+// batchSender is a function type for sending batches of journal entries.
+type batchSender func(batch []apitype.JournalEntry) error
+
+// ticker is an interface for time-based triggers.
+type ticker interface {
+	C() <-chan time.Time
+	Stop()
+	Reset(d time.Duration)
+}
+
+// realTicker wraps time.Ticker to implement the ticker interface.
+type realTicker struct {
+	*time.Ticker
+}
+
+func (t *realTicker) C() <-chan time.Time {
+	return t.Ticker.C
+}
+
+func (t *realTicker) Reset(d time.Duration) {
+	t.Ticker.Reset(d)
+}
+
 // sendBatches reads journal entries off of the entries channel and sends batches when either the maximum batch size
 // or the maximum period between batches is reached. Batches are sent sequentially.
 func sendBatches(
-	ctx context.Context,
-	client *client.Client,
-	update client.UpdateIdentifier,
-	tokenSource tokenSourceCapability,
 	maxBatchSize int,
 	period time.Duration,
 	entries <-chan saveJournalEntry,
+	sender batchSender,
+	ticker ticker,
 ) {
-	ticker := time.NewTicker(period)
-
 	results := make([]chan<- error, 0, maxBatchSize)
 	batch := make([]apitype.JournalEntry, 0, maxBatchSize)
-	flush := func() {
+	lock := sync.Mutex{}
```

Comment:
Nit: there is a cleaner syntax for zero initializing a variable.
```suggestion
	var lock sync.Mutex
```